spec/02-integration/05-proxy/01-proxy_spec.lua

```diff
@@ -129,3 +129,81 @@ describe("#stream proxy interface listeners", function()
     end
   end)
 end)
+
+for _, strategy in helpers.each_strategy() do
+  if strategy ~= "off" then
+    describe("[stream]", function()
+      local MESSAGE = "echo, ping, pong. echo, ping, pong. echo, ping, pong.\n"
+      lazy_setup(function()
+        local bp = helpers.get_db_utils(strategy, {
+          "routes",
+          "services",
+          "plugins",
+        })
+
+        local service = assert(bp.services:insert {
+          host     = helpers.mock_upstream_host,
+          port     = helpers.mock_upstream_stream_port,
+          protocol = "tcp",
+        })
+
+        assert(bp.routes:insert {
+          destinations = {
+            { port = 19000 },
+          },
+          protocols = {
+            "tcp",
+          },
+          service = service,
+        })
+
+        assert(helpers.start_kong({
+          database      = strategy,
+          stream_listen = helpers.get_proxy_ip(false) .. ":19000, " ..
+                          helpers.get_proxy_ip(false) .. ":18000, " ..
+                          helpers.get_proxy_ip(false) .. ":17000",
+          port_map      = "19000:18000",
+          plugins       = "bundled,ctx-tests",
+          nginx_conf    = "spec/fixtures/custom_nginx.template",
+          proxy_listen  = "off",
+          admin_listen  = "off",
+        }))
+      end)
```

Comment:
We should split the test into two to keep it readable for our future selves. 
With port_map enabled:
- routing based on host port works
- routing based on kong port works

And in both these cases, configure Kong to listen on a single port only.
If you are configuring kong to listen on the host_port as well, then that confuses the reader (the reader being the programmer here).

Reply:
I cannot make Kong to listen single port, as it would mean that I need to have the actual port mapping in test suite. This emulates it.